libs/zzz/solver/src/calc.ts

```diff
@@ -1,22 +1,185 @@
-import type { DiscSetKey } from '@genshin-optimizer/zzz/consts'
-import { disc2pEffect } from '@genshin-optimizer/zzz/consts'
+import { objKeyMap } from '@genshin-optimizer/common/util'
+import type {
+  AnomalyDamageKey,
+  AttributeDamageKey,
+  DiscSetKey,
+} from '@genshin-optimizer/zzz/consts'
+import {
+  allAnomalyDmgKeys,
+  allAttributeDamageKeys,
+  disc2pEffect,
+} from '@genshin-optimizer/zzz/consts'
 import type { BaseStats, DiscStats } from './common'
 
+/**
+ * sum up stats from base + discs + 2p effects
+ */
 export function getSum(baseStats: BaseStats, discs: DiscStats[]) {
   const sum = { ...baseStats }
+  const s = (key: string) => sum[key] || 0
   for (const d of discs) {
     for (const key in d.stats) {
-      sum[key] = (sum[key] || 0) + d.stats[key]
+      sum[key] = s(key) + d.stats[key]
     }
   }
   for (const [key, value] of Object.entries(sum)) {
     if (value > 2 && disc2pEffect[key as DiscSetKey])
       for (const [k, v] of Object.entries(disc2pEffect[key as DiscSetKey]))
-        sum[k] = (sum[k] || 0) + v
+        sum[k] = s(k) + v
   }
   // Rudimentary Calculations
-  sum['hp'] = (sum['hp_base'] || 0) * (1 + (sum['hp_'] || 0)) + (sum['hp'] || 0)
-  sum['atk'] = (sum['atk_base'] || 0) * (1 + (sum['atk_'] || 0))
-  sum['def'] = (sum['def_base'] || 0) * (1 + (sum['def_'] || 0))
+  sum['initial_hp'] = s('hp_base') * (1 + s('hp_')) + s('hp')
+  sum['initial_atk'] = s('atk_base') * (1 + s('atk_')) + s('atk')
+  sum['initial_def'] = s('def_base') * (1 + s('def_')) + s('def')
+  sum['final_hp'] = s('initial_hp') * (1 + s('uncond_hp_')) + s('uncond_hp')
```

Comment:
this is backwards? stats are:
`(base * unconditional% + unconditional flat) * conditional% + conditional flat)`

Reply:
You are right, it's named incorrected (unconditional -> conditional)